pxr/usd/usdShade/validators.cpp

```diff
@@ -590,6 +589,244 @@ _SubsetsMaterialBindFamily(const UsdPrim& usdPrim)
+        extension = TfStringToLower(extension);
+        static const std::unordered_set<std::string> eightBitExtensions =
+            {"bmp", "tga", "png", "jpg", "jpeg", "tif"};
+
+        return eightBitExtensions.find(extension) != eightBitExtensions.end();
+    };
+
+    if (!textureIs8Bit(textureAssetPath.GetResolvedPath())) {
+        // Nothing more is required for image depths > 8 bits, which
+        // we assume FOR NOW, are floating point
+        return errors;
+    }
+
+    TfToken colorSpace;
+    TfToken rawColorSpace("raw");
+    bool valueForColorSpaceExists = getInputValue(sourceShader, TfToken("sourceColorSpace"), colorSpace);
+    if (!valueForColorSpaceExists || colorSpace != rawColorSpace) {
+        errors.emplace_back(
+            UsdShadeValidationErrorNameTokens->invalidSourceColorSpace,
+            UsdValidationErrorType::Error,
+            UsdValidationErrorSites{
+                    UsdValidationErrorSite(usdPrim.GetStage(),
+                                           sourcePrim.GetPath())
+            },
+            TfStringPrintf("UsdUVTexture prim <%s> that reads"
+                           " Normal Map @%s@ should set "
+                           "inputs:sourceColorSpace to 'raw'.",
+                           sourcePrim.GetPath().GetText(),
+                           textureAssetPath.GetAssetPath().c_str()));
+    }
+
+    GfVec4f bias;
+    bool valueForBiasExists = getInputValue(sourceShader, TfToken("bias"), bias);
+
+    GfVec4f scale;
+    bool valueForScaleExists = getInputValue(sourceShader, TfToken("scale"), scale);
+
+    if (!(valueForBiasExists && valueForScaleExists))
+    {
+        errors.emplace_back(
+            UsdShadeValidationErrorNameTokens->nonCompliantBiasAndScale,
+            UsdValidationErrorType::Error,
+            UsdValidationErrorSites{
+                UsdValidationErrorSite(usdPrim.GetStage(), sourcePrim.GetPath())
+            },
+            TfStringPrintf("UsdUVTexture prim <%s> reads 8 bit Normal Map "
+                           "@%s@, which requires that inputs:scale be set to "
+                           "(2, 2, 2, 1) and inputs:bias be set to "
+                           "(-1, -1, -1, 0) for proper interpretation as per "
+                           "the UsdPreviewSurface and UsdUVTexture docs.",
+                           sourcePrim.GetPath().GetText(), textureAssetPath.GetAssetPath().c_str())
+        );
+        return errors;
+    }
+
+    // We still warn for inputs:scale not conforming to UsdPreviewSurface
+    // guidelines, as some authoring tools may rely on this to scale an
+    // effect of normal perturbations.
+    // don't really care about fourth components...
+    bool nonCompliantScaleValues = scale[0] != 2 ||
+        scale[1] != 2 || scale[2] != 2;
+
+    if (nonCompliantScaleValues)
+    {
+        errors.emplace_back(
+            UsdShadeValidationErrorNameTokens->nonCompliantScale,
+            UsdValidationErrorType::Warn,
```

Comment:
@tallytalwar in the compliance checker, this is a warning, however the bias check below is an error. Is that correct? I wanted to change the bias error below to be a warning, let me know what you think

Reply:
Referring the original commit for bias and scale adjustments to compliance checker: https://github.com/PixarAnimationStudios/OpenUSD/commit/b718721744f72c310f9b08aeac101c9fecd9d89a

When complying with UsdPreviewTexture and UsdUVTexture (that is when nonCompliantScaleValues is false) we need to make sure to report error for bias values. Refer these here: https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usdImaging/plugin/usdShaders/shaders/shaderDefs.usda#L100-L112

Regarding non-conforming scale values being reported as warnings: There is a reasoning here: https://github.com/PixarAnimationStudios/OpenUSD/blob/dev/pxr/usd/usdUtils/complianceChecker.py#L550-L552 (I will request to keep all these code comments as-is in the cpp so as to save the reasoning). I believe Blender is the application mentioned here (cc. @dgovil is that still the case?)

Reply:
made sure to include all comments from the complianceChecker code (I did miss one). Also rebased this branch